src/buttons/RolesEditionButton/components/RolesEditionDialog/RolesEditionDialog.test.js

```diff
@@ -0,0 +1,409 @@
+      expect(AddPeopleSelect.List).toBeNull();
+    });
+
+    test('agents that are in specificAccessByAgent are not displayed in "Add people" list', async () => {
+      setUp(propsWithUsers);
+      await AddPeopleSelect.openList();
+
+      SAMPLE_AGENTS.specificAccessByAgent.forEach((agent) => {
+        expect(AddPeopleSelect.getOption(agent.id)).toBeNull();
+      });
+    });
+
+    test('agents that are not in specificAccessByAgent are displayed in the “Add people” list', async () => {
+      setUp(propsWithUsers);
+      await AddPeopleSelect.openList();
+      agentsNotInSpecificAccess.forEach((agent) => {
+        expect(AddPeopleSelect.getOption(agent.id)).toBeInTheDocument();
+      });
+    });
+
+    test('selecting an agent in the “Add people” list opens the “New access” view in the dialog', async () => {
+      setUp(propsWithUsers);
+
+      expect(getNewAccessView()).toBeNull();
+      await openNewAccessView('user2');
+      expect(getNewAccessView()).toBeDefined();
+    });
+  });
+
+  describe('"New Access" view', () => {
+    test('"new Access" view openned with good user', async () => {
+      setUp(propsWithUsers);
+      await openNewAccessView('user2');
+
+      const expectedProps = {
+        selectedAgent: {
+          id: 'user2',
+        },
+      };
+
+      expect(mockRolesAddingDialogProps).toEqual(expect.objectContaining(expectedProps));
+    });
+
+    test('"new Access" view openned with all roles', async () => {
+      setUp(propsWithUsers);
+      await openNewAccessView('user2');
+
+      const expectedProps = {
+        allRoles: ALL_ROLES,
+      };
+
+      expect(mockRolesAddingDialogProps).toEqual(expect.objectContaining(expectedProps));
+    });
+
+    test('if preventNoneRoleForAgent, "new Access" view openned without none role', async () => {
+      setUp(propsWithPreventNoneRole);
+      await openNewAccessView('user2');
+
+      const expectedProps = {
+        allRoles: allRolesWithoutNone,
+      };
+
+      expect(mockRolesAddingDialogProps).toEqual(expect.objectContaining(expectedProps));
+    });
+  });
+
+  describe('Access control list', () => {
+    describe('Agents', () => {
+      describe('Agents displayed', () => {
+        beforeEach(() => {
+          setUp(propsWithUsers);
+        });
+
+        test('agents in specificAccessByAgent are displayed with current role', () => {
+          SAMPLE_AGENTS.specificAccessByAgent.forEach((agent) => {
+            expect(getRoleEditor(agent.id)).toBeInTheDocument();
+            expect(getCurrentRoleForAgent(agent.id)).toEqual(agent.role);
+          });
+        });
+
+        test('agents not in specificAccessByAgent are not displayed', () => {
+          agentsNotInSpecificAccess.forEach((agent) => {
+            expect(getRoleEditor(agent.id)).toBeNull();
+          });
+        });
+      });
+
+      describe('Roles passed to RoleEditor', () => {
+        test('without preventRoleForAgent AllRoles are passed to RoleEditor', () => {
```

Comment:
Typo in prop name: `preventRoleForAgent` -> `preventNoneRoleForAgents`
(idem in the following test)